File: model-api/src/main/java/io/ehdev/conrad/version/bumper/VersionBumper.java
package io.ehdev.conrad.version.bumper;

import io.ehdev.conrad.commit.CommitDetails;
import io.ehdev.conrad.version.commit.CommitVersion;

public interface VersionBumper {
    CommitVersion createNextVersion(CommitVersion parentVersion, CommitDetails commitDetails);

    CommitVersion createDefaultNextVersion(CommitVersion parentVersion);
}

File: api/version-bumper/src/main/kotlin/tech/crom/version/bumper/VersionBumper.kt
package tech.crom.version.bumper

import tech.crom.model.commit.CommitDetails
import tech.crom.model.commit.impl.RealizedCommit
import tech.crom.model.commit.impl.RequestedCommit
import tech.crom.model.commit.VersionDetails

interface VersionBumper {
    fun calculateNextVersion(commitModel: RequestedCommit, lastVersion: VersionDetails?): RealizedCommit
}
